### PR TITLE
added integration tests to github workflow action

### PR DESCRIPTION
**Pull request by sarathbrp**
Added ansible integration tests to pipeline.

### CODE REVIEW

**Review by ttuffin — COMMENTED** [review]
Are you sure these steps are necessary? The ubuntu-latest image already includes Docker.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -45,3 +45,67 @@ jobs:
...
+          path: ansible_collections/ansible/eda
+          ref: patch-1
+
+      - name: Checkout ansible/eda-server
+        uses: actions/checkout@v3
+        with:
+          repository: ansible/eda-server
+          path: eda-server # Note: Specify a separate path for this repository
+       
+      - name: Setup Docker Engine
```
Comment (ttuffin) [inline .github/workflows/tests.yml:76]: Are you sure these steps are necessary? The ubuntu-latest image already includes Docker.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -45,3 +45,67 @@ jobs:
...
+      EDA_CONTROLLER_HOST: "https://localhost:8443"
+      EDA_CONTROLLER_USERNAME: "admin"
+      EDA_CONTROLLER_PASSWORD: "testpass"
+      EDA_CONTROLLER_VERIFY_SSL: "False"
+
+    steps:
+      - name: Checkout ansible/eda
+        uses: actions/checkout@v3
+        with:
+          repository: sarathbrp/event-driven-ansible
```
Comment (kdelee) [inline .github/workflows/tests.yml:66]: should you change this to `ansible/event-driven-ansible` ?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (sarathbrp) [reply .github/workflows/tests.yml:70]: not required, removed.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -45,3 +45,56 @@ jobs:
...
         working-directory: ansible_collections/ansible/eda
+
+  integration-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version:
+          - "3.10"
+          - "3.9"
```
Comment (kdelee) [inline .github/workflows/tests.yml:56]: fwiw I think other ansible teams targeting python 3.11 only

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -45,3 +45,56 @@ jobs:
...
+        uses: actions/checkout@v3
+        with:
+          repository: ansible/eda-server
+          path: eda-server # Note: Specify a separate path for this repository
+       
+      - name: Run API in background
+        run: |
+          docker compose -p eda -f docker-compose-stage.yaml pull
+          docker compose -p eda -f docker-compose-stage.yaml up -d
+          until curl -s http://localhost:8000/_healthz | grep -q "OK"; do
```
Comment (kdelee) [inline .github/workflows/tests.yml:83]: :man_cook: nice

File: .github/workflows/tests.yml (hunk 3 above)
Comment (sarathbrp) [reply .github/workflows/tests.yml]: okay, updated

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ on:
   # there is a CI failure related to a
   # change in any dependency.
   schedule:
-    - cron: '0 */8 * * *'
+    - cron: '0 0 * * *'
```
Comment (ttuffin) [inline .github/workflows/tests.yml:12]: What is the reason for changing this?

Comment (kdelee) [reply .github/workflows/tests.yml:12]: was: every 8th hour (I'm guessing 8am and 8pm)
now: every midnight

**Review by ttuffin — APPROVED** [review]

**Review by kdelee — APPROVED** [review]

**Pull request merged**